wren-ai-service/eval/prediction.py

```diff
@@ -0,0 +1,138 @@
+        "date": datetime.now(),
+    }
+
+
+def write_prediction(meta, predictions, dir_path="outputs/predictions") -> None:
+    if Path(dir_path).exists() is False:
+        Path(dir_path).mkdir(parents=True, exist_ok=True)
+
+    output_path = f"{dir_path}/prediction_{meta['session_id']}.toml"
+
+    doc = document()
+    doc.add("meta", meta)
+    doc.add("predictions", predictions)
+
+    with open(output_path, "w") as file:
+        file.write(dumps(doc))
+
+
+def predict(meta: dict, queries: list, pipes: dict) -> List[Dict[str, Any]]:
+    predictions = []
+
+    @observe(name="Prediction Process")
+    async def wrapper(query: dict) -> None:
+        prediction = {
+            "trace_id": langfuse_context.get_current_trace_id(),
+            "trace_url": langfuse_context.get_current_trace_url(),
+            "input": query["question"],
+            "actual_output": [],
+            "expected_output": query["sql"],
+            "retrieval_context": [],
+            "context": query["context"],
+        }
+
+        # todo: do we need version, release, and other metadata?
+        langfuse_context.update_current_trace(
+            session_id=meta["session_id"],
+            user_id=meta["user_id"],
+        )
+
+        result = await pipes["retrieval"].run(query=prediction["input"])
+        documents = result.get("retrieval", {}).get("documents", [])
+        actual_output = await pipes["generation"].run(
+            query=prediction["input"],
+            contexts=documents,
+            exclude=[],
+        )
+
+        prediction["actual_output"] = actual_output
+        prediction["retrieval_context"] = [
+            {key: doc[key] for key in ("id", "content", "score")}
+            for doc in (doc.to_dict() for doc in documents)
+        ]
+
+        predictions.append(prediction)
+
+    for query in tqdm(queries, desc="Generating Predictions"):
+        asyncio.run(wrapper(query))
+
+    return predictions
+
+
+def deploy_model(mdl, pipe) -> None:
+    async def wrapper():
+        await pipe.run(orjson.dumps(mdl).decode())
+
+    asyncio.run(wrapper())
+
+
+def setup_pipes() -> Dict[str, Any]:
+    (
+        llm_provider,
+        embedder_provider,
+        document_store_provider,
+        engine,
+    ) = utils.init_providers()
+
+    document_store_provider.get_store(recreate_index=True)
+    return {
+        "indexing": indexing.Indexing(
+            embedder_provider=embedder_provider,
+            document_store_provider=document_store_provider,
+        ),
+        "retrieval": retrieval.Retrieval(
+            embedder_provider=embedder_provider,
+            document_store_provider=document_store_provider,
+        ),
+        "generation": generation.Generation(
+            llm_provider=llm_provider,
+            engine=engine,
+        ),
+    }
+
+
+def parse_args() -> Tuple[str]:
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--path", "-P", type=str, help="Path to the eval dataset")
```

Comment:
I suggest rename this argument to `--file`, `-F`, and the message says "Eval dataset file name in the `eval/dataset` folder"